cpp-sycl-ddot/ddot.sycl.cpp

```diff
@@ -49,11 +49,13 @@ int main (int argc, char* argv[]) {
   // Manual synchronization
   q.wait();
 
-  // Perform dot product using SYCL lambda kernel
+  // Perform dot product using SYCL lambda kernel and OneAPI built-in reduction
   q.submit( [&](sycl::handler& h) {
-    sycl::accessor res(bufres, h, sycl::write_only);
-    h.parallel_for(sycl::range<1>(N), [=](sycl::id<1> i) {
-      res[0] += vecA[i] * vecB[i];
+    sycl::accessor res(bufres, h, sycl::read_write);
+    auto red = sycl::ext::oneapi::reduction(res, sycl::ext::oneapi::plus<>());
```

Comment:
Reductions are part of standard SYCL 2020, I would use that instead of Intel extensions when possible.
```suggestion
    auto red = sycl::reduction(bufres, h, sycl::plus<>());
```